src/twisted/web/http.py

```diff
@@ -2947,6 +2947,9 @@ def dataReceived(self, data):
                 # itself if possible.
                 self._channel._networkProducer.unregisterProducer()
 
+                # Cancel the old channel's timeout.
+                self._channel.cancelTimeout()
+
                 transport = self._channel.transport
```

Comment:
What do you think about removing the transport from the original channel as well, for a bit of defensive programming?